.github/workflows/testsuite.yaml

```diff
@@ -84,34 +97,43 @@ jobs:
       ENV_NAME: min_deps
       PYTHON: 3.7
     name: Min Deps Testing
+    defaults:
+      run:
+        # Adding -l {0} helps ensure conda can be found properly.
+        shell: bash -l {0}
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@master
         with:
          fetch-depth: 1
-      - name: Get Miniconda Linux
-        run: |
-          wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O $HOME/miniconda.sh;
-          bash $HOME/miniconda.sh -b -p $HOME/miniconda
-      - name : Setup paths
-        run: echo "$HOME/miniconda/bin" >> $GITHUB_PATH
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2.0.0
+        with:
+          auto-update-conda: true
+          miniconda-version: "latest"
+          python-version: ${{ env.PYTHON }}
+          environment-file: ci/${{ env.ENV_NAME }}.yaml
+          activate-environment: ${{ env.ENV_NAME }}
 
-      - name: Setup Environment
+      - name: Conda Info
         run: |
-          ./ci/install_conda.sh
+          conda info -a
+          conda list
+          PYVER=`python -c "import sys; print('{:d}.{:d}'.format(sys.version_info.major, sys.version_info.minor))"`
+          if [[ $PYVER != ${{ env.PYTHON }} ]]; then
+            exit 1;
+          fi
 
       - name: Install
         run: |
-          source activate ${ENV_NAME}
           pip install .
 
       - name: Run Tests
         run: |
-          source activate $ENV_NAME
           python -m pytest --cov=pyuvsim --cov-config=.coveragerc --cov-report xml:./coverage.xml --junitxml=test-reports/xunit.xml
 
       - uses: codecov/codecov-action@master
-        if: success()
+        if: matrix.os == 'ubuntu-latest' && success()
         with:
           token: ${{secrets.CODECOV_TOKEN}} #required
           file: ./coverage.xml #optional
```

Comment:
also another little note, we can also upload coverage for other platforms by doing this
```
        name: Upload Coverage
        if: success()
        run: |
          bash <(curl -s https://codecov.io/bash) -t ${{ secrets.CODECOV_TOKEN }}
```

Reply:
I was actually just trying to make it match the full test matrix above, but I just realized we don't need it here because this is only built on linux. I'm not sure we need to upload the coverage from Mac builds, but I guess we could.

Reply:
Good point. Just an option, and written down too, if we want to do it later.

Reply:
I actually think that the codecov issues are related to this, all the coverage drops are related to running without all the dependencies, so I think the min deps upload didn't happen somehow. I'll take it out. I'm happy to add the Mac coverage, it shouldn't really take any extra time, so no harm.

Reply:
Actually, according to the readme on that action, it should work on Mac & Windows: 
https://github.com/codecov/codecov-action

Reply:
Oh that is great news! My knowledge is probably from older versions.

Reply:
I'll give it a shot...